File: App/src/main/java/com/EEA/App/models/Rental.java
/**
 * @Date: 24-12-2020
 * @Description: Describes the model
 *               class for the rental entity
 *               'Rental': An order added for a specific
 *               item of lessor by a lessee
 **/
package com.EEA.App.models;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.time.LocalDate;

@Entity
@NamedEntityGraph(name = "rental-user-item-graph",
        attributeNodes={
                @NamedAttributeNode("item"),
                @NamedAttributeNode("user")
        })
@Table(name = "rental")
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Rental extends AuditModel{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    private LocalDate rentalDate;

    @NotNull
    private LocalDate returnDate;

    @NotNull
    private Double totalPrice;

    @NotNull
    private EStatus status;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "item_id", nullable = false)
    private Item item;

    public Rental(Long id, @NotNull LocalDate rentalDate, @NotNull LocalDate returnDate,
                  @NotNull Double totalPrice, User user, Item item, EStatus status) {
        this.id = id;
        this.rentalDate = rentalDate;
        this.returnDate = returnDate;
        this.totalPrice = totalPrice;
        this.user = user;
        this.item = item;
        this.status=status;
    }

    public Rental(){}

    public EStatus getStatus() { return status; }

    public void setStatus(EStatus status) { this.status = status; }

    public Long getId() { return id; }

    public void setId(Long id) { this.id = id; }

    public void setRentalDate(LocalDate rentalDate) {
        this.rentalDate = rentalDate;
    }

    public void setReturnDate(LocalDate returnDate) {
        this.returnDate = returnDate;
    }

    public LocalDate getRentalDate() {
        return rentalDate;
    }

    public LocalDate getReturnDate() {
        return returnDate;
    }

    public Double getTotalPrice() { return totalPrice; }

    public void setTotalPrice(Double totalPrice) { this.totalPrice = totalPrice; }

    public User getUser() { return user; }

    public void setUser(User user) { this.user = user; }

    public Item getItem() { return item; }

    public void setItem(Item item) { this.item = item; }
}

File: rentalMS/src/app/entityModels/rental.ts
export enum EStatus {
  CANCELLED = 'CANCELLED',
  BOOKED = 'BOOKED',
  EXTENDED = 'EXTENDED',
  PICKED = 'PICKED',
  RETURNED = 'RETURNED',
  COMPLETED = 'COMPLETED'
}
export class Rental {

  private id: number;

  private rentalDate: Date;

  private returnDate: Date;

  private totalPrice: number;

  private status: EStatus;

  private userId: number;

  private itemId: number;
  constructor(){
  }

  get _id(): number {
    return this.id;
  }

  set _id(value: number) {
    this.id = value;
  }

  get _rentalDate(): Date {
    return this.rentalDate;
  }

  set _rentalDate(value: Date) {
    this.rentalDate = value;
  }
  get _returnDate(): Date {
    return this.returnDate;
  }

  set _returnDate(value: Date) {
    this.returnDate = value;
  }
  get _totalPrice(): number {
    return this.totalPrice;
  }

  set _totalPrice(value: number) {
    this.totalPrice = value;
  }
  get _status(): EStatus {
    return this.status;
  }

  set _status(value: EStatus) {
    this.status = value;
  }

  get _userId(): number {
    return this.userId;
  }

  set _userId(value: number) {
    this.userId = value;
  }

  get _itemId(): number {
    return this.itemId;
  }

  set _itemId(value: number) {
    this.itemId = value;
  }



}
